### PR TITLE
implementation of chain parameters migration

### LINKED ISSUE #544: Implement pool parameters migration P7->P8.
**Task description**
The migration from pool parameters from P7 to P8 needs to be implemented.

**Sub-tasks**
- [ ] subtask 1
- [ ] subtask 2
- [ ] subtask 3

*Add label for component and priority.*

### LINKED ISSUE #545: Implement chain parameters migration p7->p8
**Task description**
Implement chain parameters migration p7->p8

**Sub-tasks**
- [ ] subtask 1
- [ ] subtask 2
- [ ] subtask 3

*Add label for component and priority.*

### PR DESCRIPTION
**Pull request by drsk0**
## Purpose
Implementation of chain parameter migration for P7 -> P8.

Fixes #544.
Fixes #545.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [ ] I accept the above linked CLA.

### CODE REVIEW

File: haskell-src/Concordium/Genesis/Data/P8.hs [hunk 1]
```diff
@@ -14,6 +14,34 @@ import qualified Concordium.Crypto.SHA256 as Hash
...
+        return ProtocolUpdateData{..}
+
+-- | Parameters used to migrate state from 'P7' to 'P8'.
+data StateMigrationData = StateMigrationData
+    { -- | Data provided by the protocol update to be used
+      --  in the migration.
+      migrationProtocolUpdateData :: !ProtocolUpdateData,
+      -- | The timestamp for the first trigger block in the new consensus.
+      --  This should be set to one epoch past the regenesis time.
+      migrationTriggerBlockTime :: !Timestamp
```
Comment (eb-concordium) [inline haskell-src/Concordium/Genesis/Data/P8.hs:42]: Is this needed anywhere?

Comment (drsk0) [reply haskell-src/Concordium/Genesis/Data/P8.hs]: I assumed this is a field needed in the migration mechanism. But maybe not. I removed it, the build works. Let's see if the tests work as well.

File: haskell-src/Concordium/Genesis/Data/P8.hs [hunk 2]
```diff
@@ -14,6 +14,34 @@ import qualified Concordium.Crypto.SHA256 as Hash
...
+
+instance Serialize ProtocolUpdateData where
+    put ProtocolUpdateData{..} = do
+        put updateValidatorScoreParameters
+    get = do
+        updateValidatorScoreParameters <- get
+        return ProtocolUpdateData{..}
+
+-- | Parameters used to migrate state from 'P7' to 'P8'.
+data StateMigrationData = StateMigrationData
```
[the remark this answers is not in the document]
Comment (drsk0) [reply haskell-src/Concordium/Genesis/Data/P8.hs]: I think that's true. Are the whole hash functions and serializations correct though? This is all just copied from the P6 file.

**Review by eb-concordium — APPROVED** [review]

File: haskell-src/Concordium/Genesis/Data/P8.hs (hunk 2 above)
[the remark this answers is not in the document]
Comment (eb-concordium) [reply haskell-src/Concordium/Genesis/Data/P8.hs]: I think it looks good now

**Pull request merged**